src/main/java/seedu/address/model/task/ScheduledTask.java

```diff
@@ -30,7 +31,8 @@ protected ScheduledTask(Description description, TaskDateTime taskDateTime, Modu
 
     protected ScheduledTask(Description description, TaskDateTime taskDateTime, ModuleCode moduleCode, int taskNum) {
         this.description = description;
-        this.moduleCode = moduleCode;
+        this.moduleCode = moduleCode.getClone(); // returns a defensive copy
+        assert (taskDateTime != null) : "A ScheduledTask must have a valid time frame";
```

Comment:
If you want defensive code here, use a `requireNonNull` for all other fields. If defend, defend all

Reply:
Only `ModuleCode` has dependency with other sections in Mod Manager so it makes sense I think.